### PR TITLE
✨SDEX quote on swap

### PR DESCRIPTION
**Pull request by MattPoblete**
The goal is to get the most profitable path between SDEX & Soroswap, and use it to trade assets.

### CODE REVIEW

File: src/functions/generateRoute.ts [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
+import { Currency, CurrencyAmount, Networks, Percent, Protocols, Route, Router, Token, TradeType } from 'soroswap-router-sdk';
 import { AppContext } from 'contexts';
 import { fetchAllPhoenixPairs, fetchAllSoroswapPairs } from 'services/pairs';
+import axios from 'axios';
+import { TokenType } from 'interfaces';
+import { getBestPath, getHorizonBestPath } from 'helpers/horizon/getHorizonPath';
+import { PlatformType } from 'state/routing/types';
+import { CurrencyAmount as AmountAsset } from 'interfaces';
+
+export interface BuildTradeRoute {
```
Comment (joaquinsoza) [inline src/functions/generateRoute.ts:13]: esto deberia extenderse del BuildTradeRoute que viene de soroswap-router-sdk

Ejemplo: 

```ts
// Define the existing interface
interface OldInterface {
  property1: string;
  property2: number;
}

// Extend the existing interface
interface NewExtended extends OldInterface {
  property3: boolean;
  property4?: string; // Optional property
}
```

File: src/state/routing/types.ts [hunk 2]
```diff
@@ -232,13 +232,20 @@ export enum TradeType {
   EXACT_INPUT = 'EXACT_INPUT',
   EXACT_OUTPUT = 'EXACT_OUTPUT',
 }
+
+export enum PlatformType {
+  SOROBAN = 'Soroban',
+  STELLAR_CLASSIC = 'SDEX'
+}
```
Comment (joaquinsoza) [inline src/state/routing/types.ts:239]: Should be [Router, Sdex] to then be able to add Aggregator

**Review by joaquinsoza — CHANGES_REQUESTED** [review]

File: src/constants/tokens.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+export const tokens = {
```
[the remark this answers is not in the document]
Comment (MattPoblete) [reply src/constants/tokens.ts:1]: Una lista de tokens utilizada durante el desarrollo en testnet.
Se creó porque la respuesta de la API de soroswap no posee la propiedad "issuer"

File: src/functions/generateRoute.ts (hunk 1 above)
Comment (MattPoblete) [reply src/functions/generateRoute.ts:13]: La clase "BuildTradeReturn" de soroswap-router-sdk presenta incompatibilidades de tipo en propiedades que son necesarias para el funcionamiento de la nueva logica:
"Interface 'BuildTradeRoute' incorrectly extends interface 'BuildTradeReturn'.
  Types of property 'amountCurrency' are incompatible.
    Type 'CurrencyAmount | CurrencyAmount<Currency>' is not assignable to type 'CurrencyAmount'.
      Type 'CurrencyAmount' is missing the following properties from type 'CurrencyAmount<Currency>': decimalScale, add, subtract, multiply, and 14 more"

**Comment by joaquinsoza** [comment]
Merge errors

**Review by chopan123 — APPROVED** [review]

**Pull request merged**